src/Analyzers/CSharp/Tests/MakeMemberStatic/MakeMemberStaticTests.cs

```diff
@@ -129,5 +144,152 @@ public static class Foo
+                "abstract, sealed, override, readonly, unsafe, volatile, async";
+
+            await TestAsync(testCode, fixedCode, customModifierOrder);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMemberStatic)]
+        public async Task TestCustomOrderEnd()
+        {
+            var testCode = @"
+public static class Foo
+{
+    public async System.Threading.Tasks.Task {|CS0708:Test|}() { }
+}
+";
+
+            var fixedCode = @"
+public static class Foo
+{
+    public async static System.Threading.Tasks.Task Test() { }
+}
+";
+            var customModifierOrder = "public, private, protected, internal, extern, new, virtual, abstract, " +
+                "sealed, override, readonly, unsafe, volatile, async, static";
+
+            await TestAsync(testCode, fixedCode, customModifierOrder);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMemberStatic)]
+        public async Task TestWrongInitialOrder()
+        {
+            var testCode = @"
+public static class Foo
+{
+    async public System.Threading.Tasks.Task {|CS0708:Test|}() { }
+}
+";
+
+            var fixedCode = @"
+public static class Foo
+{
+    async public static System.Threading.Tasks.Task Test() { }
+}
+";
+
+            await TestAsync(testCode, fixedCode);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMemberStatic)]
+        public async Task TestTriviaBetweenModifiers()
+        {
+            var testCode = @"
+public static class Foo
+{
+    public /* trivia */ async System.Threading.Tasks.Task {|CS0708:Test|}() { }
+}
+";
+
+            var fixedCode = @"
+public static class Foo
+{
+    public /* trivia */ static async System.Threading.Tasks.Task Test() { }
+}
+";
+
+            await TestAsync(testCode, fixedCode);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMemberStatic)]
+        public async Task TestNoOtherModifiers()
+        {
+            var testCode = @"
+public static class Foo
+{
+    void {|CS0708:Test|}() { }
+}
+";
+
+            var fixedCode = @"
+public static class Foo
+{
+    static void Test() { }
+}
+";
+
+            await TestAsync(testCode, fixedCode);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMemberStatic)]
+        public async Task TestNoStaticDefined()
+        {
+            var testCode = @"
+public static class Foo
+{
+    public async System.Threading.Tasks.Task {|CS0708:Test|}() { }
+}
+";
+
+            var fixedCode = @"
+public static class Foo
+{
+    public async static System.Threading.Tasks.Task Test() { }
+}
+";
+
+            var customModifierOrder = "public, private, protected, internal, extern, new, virtual, abstract, " +
+                "sealed, override, readonly, unsafe, volatile, async";
+
+            await TestAsync(testCode, fixedCode, customModifierOrder);
+        }
```

Comment:
would be good to have some tests with interesting arrangements of static and the other modifiers (including where the other modifiers are not specified) and where there are interesting different sets of preceding modifers in teh actual code.

Reply:
Out of all the modifiers C# has, there's only a handful that can be applied to static members so they would work together, regardless I added some more complex scenarios to the tests.